hudi-common/src/main/java/org/apache/hudi/common/table/log/HoodieMetadataMergedLogRecordScanner.java

```diff
@@ -78,24 +78,27 @@ public Map<String, HoodieRecord> getRecords() {
 
   @Override
   public <T> void processNextRecord(HoodieRecord<T> newRecord) throws IOException {
-    // Merge the new record with the existing record in the map
-    HoodieRecord<T> oldRecord = (HoodieRecord<T>) records.get(newRecord.getRecordKey());
-    if (oldRecord != null) {
-      LOG.debug("Merging new record with existing record in the map. Key: {}", newRecord.getRecordKey());
-      recordMerger.fullOuterMerge(oldRecord, readerSchema, newRecord, readerSchema, this.getPayloadProps()).forEach(
-          mergedRecord -> {
-            HoodieRecord<T> combinedRecord = mergedRecord.getLeft();
-            if (combinedRecord.getData() != oldRecord.getData()) {
-              HoodieRecord latestHoodieRecord = getLatestHoodieRecord(newRecord, combinedRecord, newRecord.getRecordKey());
-              records.put(newRecord.getRecordKey(), latestHoodieRecord.copy());
-            }
-          });
+    String key = newRecord.getRecordKey();
```

Comment:
The logic for processNextRecord is now same as HoodieMergedLogRecordScanner, so we won't need separate HoodieMetadataMergedLogRecordScanner. However, I would prefer to do this cleanup in a separate patch.

Reply:
I wanted to keep it simple for review. I can do it in this patch itself.